models/repo.go

```diff
@@ -470,6 +470,12 @@ func MigrateRepository(u *User, name, desc string, private, mirror bool, url str
 		return repo, fmt.Errorf("create update hook: %v", err)
 	}
 
+	// Check if repository is empty.
+	_, _, err = com.ExecCmdDir(repoPath, "git", "rev-parse", "--verify", "--quiet", "HEAD")
```

Comment:
OK, so if you don't use `--quite`, you can check error string for `fatal: Needed a single revision`.